server/xpub-controller/manuscript.js

```diff
@@ -1,17 +1,195 @@
 const logger = require('@pubsweet/logger')
 const ManuscriptModel = require('@elifesciences/xpub-model').Manuscript
+const FileModel = require('@elifesciences/xpub-model').File
 const UserModel = require('@elifesciences/xpub-model').User
+
 const Notification = require('./notification')
 
 class Manuscript {
-  constructor(config, user) {
+  constructor(config, user, storage, scienceBeamApi, pubsubManager) {
     this.user = user
     this.config = config
+    this.storage = storage
+    this.scienceBeamApi = scienceBeamApi
+    this.pubsubManager = pubsubManager
+  }
+
+  async upload(manuscriptId, file, fileSize) {
+    const { ON_UPLOAD_PROGRESS } = this.pubsubManager.asyncIterators
+
+    const userId = await UserModel.getProfileForUuid(this.user)
```

Comment:
This means another call to the database, which I'm not adverse to.

What's your opinion on https://github.com/elifesciences/elife-xpub/pull/1474#discussion_r255058801